subprojects/shipkit/src/main/groovy/org/shipkit/internal/util/GitHubStatusCheck.java

```diff
@@ -17,43 +18,48 @@
     private MergePullRequestTask task;
     private GitHubApi gitHubApi;
     private final int amountOfRetries;
+    private final long defaultInitialTimeout;
 
-
-
-    public GitHubStatusCheck(MergePullRequestTask task, GitHubApi gitHubApi, int amountOfRetries) {
+    public GitHubStatusCheck(MergePullRequestTask task, GitHubApi gitHubApi, int amountOfRetries, long defaultInitialTimeout) {
         this.task = task;
         this.gitHubApi = gitHubApi;
         this.amountOfRetries = amountOfRetries;
+        this.defaultInitialTimeout = defaultInitialTimeout;
     }
 
     public GitHubStatusCheck(MergePullRequestTask task, GitHubApi gitHubApi) {
         this.task = task;
         this.gitHubApi = gitHubApi;
         this.amountOfRetries = 20;
+        this.defaultInitialTimeout = TimeUnit.SECONDS.toMillis(10);
     }
 
     public PullRequestStatus checkStatusWithRetries() throws IOException, InterruptedException {
         int timeouts = 0;
         while (timeouts < amountOfRetries) {
             JsonObject status = getStatusCheck(task, gitHubApi);
-            if (status.getCollection("statuses") == null || status.getCollection("statuses").size() == 0) {
+            if (timeouts > 0 && isNullOrEmpty(status, "statuses")) {
                 return PullRequestStatus.NO_CHECK_DEFINED;
-            }
-
-            if (allStatusesPassed(status)) {
+            } else if (!isNullOrEmpty(status, "statuses") && allStatusesPassed(status)) {
```

Comment:
We're missing a scenario here: what if the check fails? If it fails, we should return failure rather than attempt to retry with timeout. I know that this code was like that before :) We can fix it in the next round.

Reply:
yep, we definitely should do so, thx!